src/DateTime.cc

```diff
@@ -436,6 +436,9 @@ DateTime::DateTime(std::string const& iso8601, Timescale scale) {
 }
 
 double DateTime::get(DateSystem system, Timescale scale) const {
+    if (!isValid()) {
+        throw LSST_EXCEPT(pexEx::RuntimeError, "DateTime not valid");
```

Comment:
My one worry is that the exception will show a less useful line number, but I think it's still worth doing (and cleaner than using a macro).